ctrd/utils.go

```diff
@@ -16,14 +16,16 @@ import (
 	"github.com/pkg/errors"
 )
 
-func resolver(authConfig *types.AuthConfig) (remotes.Resolver, error) {
+// pushTracker records push progress detail
+var pushTracker = docker.NewInMemoryTracker()
+
+func resolver(authConfig *types.AuthConfig, resolverOpt docker.ResolverOptions) (remotes.Resolver, error) {
```

Comment:
Can we make the second passed parameter an interface, rather than a specific docker resolver options?

Reply:
We can directly use docker package, I do not think wrapper this is meaningful